src/Control/Middleware/CanonicalURLMiddleware.php

```diff
@@ -349,6 +349,10 @@ public function setEnabledEnvs($enabledEnvs)
      */
     protected function isEnabled()
     {
+        if (Director::is_cli()) {
```

Comment:
This will prevent unit tests from running the middleware. See https://github.com/silverstripe/silverstripe-framework/pull/8158 for a fix that works around this.